packages/material-ui/src/Popover/Popover.js

```diff
@@ -62,7 +62,13 @@ function getScrollParent(parent, child) {
 }
 
 function getAnchorEl(anchorEl) {
-  return typeof anchorEl === 'function' ? anchorEl() : anchorEl;
+  if (typeof anchorEl === 'function') {
+    return anchorEl();
+  }
+  if (typeof anchorEl === 'object' && anchorEl && anchorEl.current) {
+    return anchorEl.current;
```

Comment:
@Andarist That's what I was talking about. Should we wrap this in `React.findDOMNode` or expect users to pass a ref to a DOM node i.e. `getAnchorEl(anchorEl: HTMLElement | React.RefObject<HTMLElement>): HTMLElement` instead of `getAnchorEl(anchorEl: HTMLElement | React.RefObject<any>): HTMLElement`

Reply:
You can still use it currently as a compatibility bridge since it won't trigger any warnings if it's called with a DOM node. So instead of throwing at runtime if people pass a ref to a component it would only trigger warnings in strict mode. However since `getAnchorEl` is called during the render method we can't use it here.

It would help a lot if you could add an example that's just a rewrite of another example while also being equivalent and "simpler".

However https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/Popover/Popover.js#L311 should also be adjusted since this would always use the `document` as a container even if `anchorEl.current` is `null`.

Reply:
> findDOMNode is being deprecated in StrictMode

As far as I know, it's not. `findDOMNode` is only deprecated in strict mode when a component instance is provided as first argument.